test/unit/test_providers_kms_master_key_provider.py

```diff
@@ -32,6 +33,7 @@ def test_init_with_regionless_key_ids_and_region_names():
 class TestKMSMasterKeyProvider(object):
     @pytest.fixture(autouse=True)
     def apply_fixtures(self):
+        self.botocore_no_region_session = botocore.session.Session(session_vars={"region": (None, None, None, None)})
```

Comment:
This syntax is weird -- the reason it wants this tuple of values for region is a little obscure, but it's due to how botocore's `session.py` unpacks the config values in `_update_config_store_from_session_vars`. I tried a bunch of variants of this and this is what worked, but I'm not sure what the right convention is to comment this bit of weirdness. Super open to feedback here.

Reply:
I don't love it either, but as we discussed offline, I don't know that there's a better way to do it. This is fine.